app/assets/javascripts/components/ReportFilter.jsx

```diff
@@ -55,65 +72,82 @@ class ReportFilter extends React.Component {
-      view_level: event.target.value
-    });
+  updateThreshold(e) {
+    const { innerText, className } = e.target;
+    if (className.indexOf('nt_zscore') >= 0) {
+      const ntZscore = document.getElementById('nt_zscore');
+      if (ntZscore) {
+        (className.indexOf('start') >= 0) 
+        ? ntZscore.noUiSlider.set([innerText, null]) : ntZscore.noUiSlider.set([null, innerText]);
+      }
+    } else if (className.indexOf('nt_rpm') >= 0) {
+      const ntRpm = document.getElementById('nt_rpm');
+      if (ntRpm) {
+        (className.indexOf('start') >= 0) 
+        ? ntRpm.noUiSlider.set([innerText, null]) : ntRpm.noUiSlider.set([null, innerText]);
+      }
+    }
   }
 
   componentDidMount() {
-    const nt_zscore = document.getElementById('nt_zscore');
-    const nt_rpm = document.getElementById('nt_rpm');
-
-    noUiSlider.create(nt_zscore, {
-      start: [this.state.nt_zscore_start, this.state.nt_zscore_end],
-      connect: true,
-      step: 1,
-      orientation: 'horizontal', // 'horizontal' or 'vertical',
-      behaviour: 'tap-drag',
-      range: {
-        'min': 0,
-        'max': this.props.highest_tax_counts.highest_nt_zscore
-      },
-      format: wNumb({
-        decimals: 0
-      })
-    });
-    noUiSlider.create(nt_rpm, {
-      start: [this.state.nt_rpm_start, this.state.nt_rpm_end],
-      connect: true,
-      step: 1,
-      orientation: 'horizontal', // 'horizontal' or 'vertical'
-      range: {
-        'min': 0,
-        'max': this.props.highest_tax_counts.highest_nt_rpm
-      },
-      format: wNumb({
-        decimals: 0
-      })
-    });
-    nt_zscore.noUiSlider.on('update', (values, handle) => {
-      if (handle === 0) {
-        this.setState({
-          nt_zscore_start: Math.round(values[handle])
-        });
-      } else {
-        this.setState({
-          nt_zscore_end: Math.round(values[handle])
-        });
-      }
-    });
-    nt_rpm.noUiSlider.on('update', (values, handle) => {
-      if (handle === 0) {
-        this.setState({
-          nt_rpm_start: Math.round(values[handle])
-        });
-      } else {
-        this.setState({
-          nt_rpm_end: Math.round(values[handle])
-        });
-      }
-    });
+
+    if (this.has_valid_nt_zscore_range) {
+      const nt_zscore = document.getElementById('nt_zscore');
+      noUiSlider.create(nt_zscore, {
+        start: [this.state.nt_zscore_start, this.state.nt_zscore_end],
+        connect: true,
+        step: 1,
+        orientation: 'horizontal', // 'horizontal' or 'vertical',
+        behaviour: 'tap-drag',
+        range: {
+          min: this.props.highest_tax_counts.lowest_nt_zscore,
+          max: this.props.highest_tax_counts.highest_nt_zscore
+        },
+        format: wNumb({
+          decimals: 0
+        })
+      });
+
+      nt_zscore.noUiSlider.on('update', (values, handle) => {
+        if (handle === 0) {
+          this.setState({
+            nt_zscore_start: Math.round(values[handle])
```

Comment:
start and end shouldn't be rounded, because data points may end up outside the rounded range

to convert `start` and `end` to integers, use the function `Math.floor` for `start`, and `Math.ceil` for end
(i am assuming JS has those, and I am also assuming that `Math.floor(-1.1)` is `-2.0` and not `-1.0` -- could you please confirm that).

But to the broader question, why are you converting to integers?   If the actual data range is [-4.35 to -1.21], shouldn't users be permitted to select values in that range?

My suggestion is for now if you need to convert to integers, to use floor and ceil, and also research whether converting to integers is wise.